### PR TITLE
forbid-component-props: add component whitelist

### LINKED ISSUE #1732: forbid-component-props should have a component whitelist
For custom components it makes sense to have the rule, but when using third-party components violating the rule (`ReactModal` using `className` is probably one of the more prominent examples) there isn't much you can do about it...

Having to add `eslint-disable react/forbid-component-props` comments isn't a very pretty solution.

### PR DESCRIPTION
**Pull request by ThiefMaster**
Docs are still missing ;)

fixes #1732

### CODE REVIEW

File: lib/rules/forbid-component-props.js [hunk 1]
```diff
@@ -31,20 +31,40 @@ module.exports = {
...
-            type: 'string'
+            oneOf: [{
+              type: 'string'
+            }, {
+              type: 'object',
+              properties: {
+                propName: {
+                  type: 'string'
+                },
+                allowedFor: {
```
Comment (ljharb) [inline lib/rules/forbid-component-props.js]: let's make sure this list is unique

File: lib/rules/forbid-component-props.js [hunk 2]
```diff
@@ -31,20 +31,40 @@ module.exports = {
...
+      }
     }]
   },
 
   create: function(context) {
-    function isForbidden(prop) {
-      const configuration = context.options[0] || {};
+    const configuration = context.options[0] || {};
+    const forbid = new Map((configuration.forbid || DEFAULTS).map(value => {
+      const propName = (typeof value === 'string') ? value : value.propName;
```
Comment (ljharb) [inline lib/rules/forbid-component-props.js]: let's remove the redundant parens on the ternary conditionals here and below

File: lib/rules/forbid-component-props.js (hunk 1 above)
Comment (ThiefMaster) [reply lib/rules/forbid-component-props.js]: done; didn't know json schema has an option to require uniqueness

**Comment by ThiefMaster** [comment]
Docs added, changes done.

**Comment by ThiefMaster** [comment]
Anything missing to get this merged? Would be nice if we could start using it (by installing the plugin from this git repo until it's part of a release, without having to maintain our own branch containing this and my other PR).

**Pull request merged**